src/Lykke.Job.TransactionHandler/Handlers/EthereumCoreCommandHandler.cs

```diff
@@ -0,0 +1,468 @@
+                        break;
+
+                    default:
+                        break;
+                }
+
+                return CommandHandlingResult.Ok();
+            }
+            catch (Exception e)
+            {
+                _log.WriteError(command.TransactionHash ?? "Empty", command, e);
+                throw;
+            }
+        }
+
+        public async Task<CommandHandlingResult> Handle(ProcessCoinEventCommand command, IEventPublisher eventPublisher)
+        {
+            try
+            {
+                switch (command.CoinEventType)
+                {
+                    case CoinEventType.CashinCompleted:
+                        await ProcessCashIn(command, eventPublisher);
+                        break;
+                    case CoinEventType.TransferCompleted:
+                    case CoinEventType.CashoutCompleted:
+                        await ProcessOutcomeOperation(command);
+                        break;
+                    case CoinEventType.CashoutFailed:
+                        await ProcessFailedCashout(command, eventPublisher);
+                        break;
+                    default:
+                        break; ;
+                }
+
+                return CommandHandlingResult.Ok();
+            }
+            catch (Exception e)
+            {
+                _log.WriteError(command.TransactionHash ?? "Empty", command, e);
+                throw;
+            }
+}
+
+        public async Task<CommandHandlingResult> Handle(EnrollEthCashinToMatchingEngineCommand command, IEventPublisher eventPublisher)
+        {
+            try
+            {
+                var cashinId = command.CashinOperationId.ToString("N");
+                var clientId = command.ClientId;
+                var hash = command.TransactionHash;
+                var amount = command.Amount;
+                var asset = await _assetsServiceWithCache.TryGetAssetAsync(command.AssetId);
+                var createPendingActions = command.CreatePendingActions;
+                var clientAddress = command.ClientAddress;
+                var paymentTransaction = PaymentTransaction.Create(hash,
+                    CashInPaymentSystem.Ethereum, clientId, (double)amount,
+                    asset.DisplayId ?? asset.Id, status: PaymentStatus.Processing);
+
+                var exists = await _paymentTransactionsRepository.CheckExistsAsync(paymentTransaction);
+
+                if (exists)
+                {
+                    _log.WriteWarning(command.TransactionHash ?? "Empty", command,
+                            $"Transaction already handled {hash}");
+
+                    return CommandHandlingResult.Ok();
+                }
+
+                if (createPendingActions)
+                {
+                    if (asset.IsTrusted)
+                    {
+                        await _ethererumPendingActionsRepository.CreateAsync(clientId, Guid.NewGuid().ToString());
```

Comment:
This call not idempotent, so, if ```_matchingEngineClient.CashInOutAsync``` call is failed, then ```_ethererumPendingActionsRepository.CreateAsyn``` will be called more then once for the same cashin. Is this permitted? Not looks so

Reply:
It is actually permitted

Reply:
Guid.NewGuid() for every new call - is it ok?